### DISCUSSION TITLE
Missing Object Name

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#139] [General]
Hello,

394 ❘ CHAPTER 15 Dependency InjectIon And Configuration

You forgot to write the name of the object for ServiceCollection class. 

```
static ServiceProvider GetServiceProvider()
{
ServiceCollection = new();  // It should be ServiceCollection services = new();  
services.AddSingleton<IGreetingService, GreetingService>();
services.AddTransient<HomeController>();
return services.BuildServiceProvider();
}
```

**Comment by christiannagel** [comment]
Thanks for reporting this. I've added an update https://github.com/ProfessionalCSharp/ProfessionalCSharp2021/blob/main/Updates.md
